fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckInfoCard.kt

```diff
@@ -158,37 +179,45 @@ enum class ReviewQualityCheckInfoType {
             Error -> FirefoxTheme.colors.actionError
             Info -> FirefoxTheme.colors.actionInfo
             AnalysisUpdate -> FirefoxTheme.colors.actionSecondary
+            Loading -> FirefoxTheme.colors.actionSecondary
         }
 
     val buttonTextColor: Color
         @Composable
         get() = when {
             this == Info && !isSystemInDarkTheme() -> FirefoxTheme.colors.textOnColorPrimary
-            this == AnalysisUpdate -> FirefoxTheme.colors.textActionSecondary
+            this == AnalysisUpdate || this == Loading -> FirefoxTheme.colors.textActionSecondary
             else -> FirefoxTheme.colors.textPrimary
         }
 }
 
-private class PreviewModel(
-    val messageType: ReviewQualityCheckInfoType,
-    val iconId: Int,
+/**
+ * Model for the optional button in a [ReviewQualityCheckInfoCard].
+ *
+ * @param text The text to show in the button.
+ * @param onClick The callback to invoke when the button is clicked.
+ */
+data class InfoCardButtonText(
+    val text: String,
+    val onClick: () -> Unit,
 )
 
-private class PreviewModelParameterProvider : PreviewParameterProvider<PreviewModel> {
-    override val values: Sequence<PreviewModel>
+private class PreviewModelParameterProvider : PreviewParameterProvider<ReviewQualityCheckInfoType> {
+    override val values: Sequence<ReviewQualityCheckInfoType>
         get() = sequenceOf(
-            PreviewModel(ReviewQualityCheckInfoType.Warning, R.drawable.mozac_ic_warning_fill_24),
-            PreviewModel(ReviewQualityCheckInfoType.Confirmation, R.drawable.mozac_ic_checkmark_24),
-            PreviewModel(ReviewQualityCheckInfoType.Error, R.drawable.mozac_ic_information_fill_24),
-            PreviewModel(ReviewQualityCheckInfoType.Info, R.drawable.mozac_ic_information_fill_24),
-            PreviewModel(ReviewQualityCheckInfoType.AnalysisUpdate, R.drawable.mozac_ic_information_fill_24),
+            ReviewQualityCheckInfoType.Warning,
+            ReviewQualityCheckInfoType.Confirmation,
+            ReviewQualityCheckInfoType.Error,
+            ReviewQualityCheckInfoType.Info,
+            ReviewQualityCheckInfoType.AnalysisUpdate,
+            ReviewQualityCheckInfoType.Loading,
```

Comment:
you might be able to now do `ReviewQualityCheckInfoType.values.asSequence()`

Reply:
yes, good call! 👌